### PR TITLE
Add missing writer.finish() in annotate and alias commands

### PR DESCRIPTION
**Pull request by kimrutherford**
This fixes invalid JSON output from the annotate and alias commands.  The closing "]" was missing with "-O json".

Both of these commands produce invalid JSON:

```sh
runoak -i bioportal: annotate -O json "enlarged nucleus in T-cells from peripheral blood"
```

```sh
runoak -i sqlite:obo:fypo aliases -O json FYPO:0009021
```

This PR is my guess at a fix.

### CODE REVIEW

**Comment by kimrutherford** [comment]
I've added more commits that add `writer.finish()` to `mappings()`, `ancestors()`, `descendants()` and `similarity_pair()` in `cli.py`.

Thanks for the great project!  I think it will be very useful for PomBase.

**Comment by cmungall** [comment]
@hrshdhgd do you know what is going on here?

```
=================================== FAILURES ===================================
_______________ TestCommandLineInterface.test_mappings_curie_map _______________

self = <tests.test_cli.TestCommandLineInterface testMethod=test_mappings_curie_map>

    def test_mappings_curie_map(self):
        mappings_output = OUTPUT_DIR.joinpath("test_mappings.tsv")
        if os.name == "nt":
            shell = True
        else:
            shell = False
        result = subprocess.run(
            [
                "runoak",
                "-i",
                f"sqlite:{TEST_DB}",
                "mappings",
                "-O",
                "sssom",
                "-o",
                mappings_output,
            ],
            shell=shell,  # noqa
        )
        self.assertEqual(0, result.returncode)
>       msdf = parse_sssom_table(mappings_output)

tests/test_cli.py:418: 
```

File: src/oaklib/cli.py [hunk 1]
```diff
@@ -3466,6 +3470,7 @@ def mappings(terms, maps_to_source, autolabel: bool, output, output_type):
                 if autolabel:
                     impl.inject_mapping_labels([mapping])
                 writer.emit(mapping)
+    writer.finish()
```
Comment (hrshdhgd) [inline src/oaklib/cli.py:3473]: @cmungall : Addition of this line creates the error you see in gh actions

Comment (hrshdhgd) [reply src/oaklib/cli.py:3473]: The output file is blank and hence the file parsed back contains nothing.

Comment (cmungall) [reply src/oaklib/cli.py:3473]: can you go ahead and make the change?

Comment (cmungall) [reply src/oaklib/cli.py:3473]: oh I see this is on a fork... we can get @kimrutherford set up with permissions to make branches off origin if that fits out flow better

**Comment by kimrutherford** [comment]
> we can get @kimrutherford set up with permissions to make branches off origin if that fits out flow better

I'm happy to help.  If that's easier for you, please add me.

I've just added a minimal test for `runoak mappings` using JSON as the output format.

**Comment by hrshdhgd** [comment]
I have added you @kimrutherford , kindly confirm!

**Comment by kimrutherford** [comment]
> I have added you @kimrutherford , kindly confirm!

Thanks.  I've rebased my changes and made a branch in the main repo:
https://github.com/INCATools/ontology-access-kit/tree/kmr-json-write-fix

If you'd like me to close this PR and make a new one with the new branch, let me know.

**Comment by hrshdhgd** [comment]
I think making a new PR with a branch within this repo would be helpful. That way I could quickly fix the obvious errors in the GitHub Actions QC and merge the branch asap.

**Comment by hrshdhgd** [comment]
I think making a new PR with a branch within this repo would be helpful. That way I could quickly fix the obvious errors in the GitHub Actions QC and merge the branch asap.

**Comment by kimrutherford** [comment]
I've open #501 to replace this PR.

**Pull request closed without merging**